presto-manager-common/src/main/java/com/teradata/prestomanager/common/ApiRequester.java

```diff
@@ -48,7 +48,7 @@ private ApiRequester(Client client,
             String mediaType)
     {
         this.client = requireNonNull(client);
-        this.uriTemplate = requireNonNull(uriBuilder).toTemplate();
+        this.uriTemplate = new UriCopyBuilder(requireNonNull(uriBuilder));
```

Comment:
It restricts the valid operations on the template to `template.clone().uri(URI).build()`, so that the `ApiRequester`'s template is guaranteed to be immutable and thread-safe to use.

I could change it to just call `clone` and name it something like `UriTemplateProvider`. Then `ApiRequester` could still be internally thread-safe, but allow later operations to use other URI changes. (I can't think of a case where that would be a good idea, but it would also make the intent a bit clearer.)